packages/opentelemetry/lib/config/metrics.js

```diff
@@ -0,0 +1,62 @@
+const {
+	OTLPMetricExporter
+} = require('@opentelemetry/exporter-metrics-otlp-proto');
+const { CompressionAlgorithm } = require('@opentelemetry/otlp-exporter-base');
+require('@opentelemetry/sdk-node');
+const { PeriodicExportingMetricReader } =
+	require('@opentelemetry/sdk-node').metrics;
+const logger = require('@dotcom-reliability-kit/logger');
+const { METRICS_USER_AGENT } = require('./user-agents');
+
+/**
+ * @typedef {object} MetricsOptions
+ * @property {string} [apiGatewayKey]
+ *     The API key to send to the metrics collector if you're using the FT's official metrics collector endpoint.
+ * @property {string} [endpoint]
+ *     The URL to send OpenTelemetry metrics to, for example http://localhost:4318/v1/metrics.
+ */
+
+/**
+ * Create an OpenTelemetry metrics configuration.
+ *
+ * @param {MetricsOptions} options
+ * @returns {Partial<import('@opentelemetry/sdk-node').NodeSDKConfiguration>}
+ */
+exports.createMetricsConfig = function createMetricsConfig(options) {
+	/** @type {Partial<import('@opentelemetry/sdk-node').NodeSDKConfiguration>} */
+	const config = {};
+
+	// If we have an OpenTelemetry metrics endpoint then set it up
+	if (options?.endpoint) {
+		const headers = {
+			'user-agent': METRICS_USER_AGENT
+		};
+		if (options.apiGatewayKey) {
+			headers['X-OTel-Key'] = options.apiGatewayKey;
+		}
+		config.metricReader = new PeriodicExportingMetricReader({
+			exporter: new OTLPMetricExporter({
+				url: options.endpoint,
+				compression: CompressionAlgorithm.GZIP,
+				headers
+			})
+		});
+
+		logger.info({
+			event: 'OTEL_METRICS_STATUS',
+			message: `OpenTelemetry metrics are enabled and exporting to endpoint ${options.endpoint}`,
+			enabled: true,
+			endpoint: options.endpoint
+		});
+	} else {
+		logger.info({
+			event: 'OTEL_METRICS_STATUS',
+			message:
+				'OpenTelemetry metrics are disabled because no metrics endpoint was set',
+			enabled: false,
+			endpoint: null
+		});
```

Comment:
**note:** I decided to switch this out from a warning because it could be a bit spammy. I think if metrics aren't being received then it's up to our Grafana config to alert rather than having something in the logs. Something I could add is an error log if:

* The OpenTelemetry package is installed
* Both tracing _and_ metrics endpoints are not configured
* The environment is `production`

This feels like a definite misconfiguration and might help alert us to badly configured apps?

Reply:
Oh that's interesting though, it'd definitely be possible for this library to just send a new metric on a timer while the process is running, might be a bit spammy though? If we did something like that we'd probably want to discuss naming etc